include/Reader/readerir.h

```diff
@@ -1228,6 +1221,32 @@ class GenIR : public ReaderBase {
                                   CorInfoType *CorType,
                                   ReaderAlignType *Alignment);
 
+  /// Check whether access to this multidimensional array can be expanded as an
+  /// intrinsic.
+  ///
+  /// \param Sig Intrinsic signature.
+  /// \param IsStore true iff this array access is a store.
+  /// \param IsLoadAddr true iff this array access is a load of an element
+  /// address.
+  /// \param Rank [OUT] Array rank.
+  /// \param ElemCorType [OUT] CorType of the array element.
+  /// \param ElemType [OUT] Type of the array element.
+  /// \returns true iff access to this multidimensional array can be expanded as
+  /// an intrinsic.
+  bool canExpandMDArrayRef(CORINFO_SIG_INFO *Sig, bool IsStore, bool IsLoadAddr,
+                           uint32_t *Rank, CorInfoType *ElemCorType,
+                           llvm::Type **ElemType);
+
+  /// Get address of an element of a multidimensional array.
+  ///
+  /// This method reads array address and indices for each dimensions from the
+  /// operand stack.
```

Comment:
Probably worth mentioning here that it uses an expansion that assumes the array pointer is non-null (and that callers need to generate a null check before calling this if the array isn't known to be non-null).